sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -286,6 +288,9 @@ object Decimal {
   /** Maximum number of decimal digits a Long can represent */
   val MAX_LONG_DIGITS = 18
 
+  /** Maximum precision a Decimal can support */
+  val MAX_PRECISION = 38
```

Comment:
ok. however, with the customized `MathContext` approach, we can not determine the needed precision needed for later decimal operations like multiplication without losing precision reported in the JIRA. Directly using java decimal to do these operations can avoid this problem as showed in first commit. Do we want to use it?